collect_app/src/main/java/org/odk/collect/android/http/HttpClientConnection.java

```diff
@@ -228,7 +229,7 @@ HttpGetResult executeGetRequest(@NonNull URI uri, @Nullable final String content
             }
         }
 
-        return new HttpGetResult(downloadStream, responseHeaders, hash, statusCode);
+        return new HttpGetResult(downloadStream, responseHeaders, hash, statusCode, (CloseableHttpClient) httpclient);
```

Comment:
Why do you use the new constructor only here but not in other classes which implements `OpenRosaHttpInterface`?

Reply:
We use Httpclient in other methods in this class as well but you edited only this one, why?

Reply:
These warnings were resolved based on the detections reported by strict mode.

Reply:
Maybe you just used that one method testing the app manually. It would be strange that in one place `httpclient` should be closed but in another place not. Could you investigate it deeper?

Reply:
I thought of the same, but then I noticed that this will soon be replaced with OkHttp implementation. So, we should not spend too much time improving obsolete codebase. Moreover, if cursor is actually not getting closed, then we will get notified for the debug builds.

Do you think it's worth investigating deeper into this given the above statements?

Reply:
Let's close it elsewhere as well. Probably won't take much time. :+1:

Reply:
Just noticed that OkHttp implemententation also uses the same Closeable response. So it is gonna happen in the new code as well

Reply:
On deeper investigation I found out that `CloseableHttpClient` is auto-closed at the end of `execute` method. There is some deeper issue with the custom http client implementation. The auto-close doesn't work. Even calling the `close()` method explicitly doesn't work at all. On the contrary, with `OkHttpConnection` class, I observed no such cursor leak even without explicitly calling the `close()` method.

So, let's merge this PR only only after #2869 is resolved otherwise the debug app will continuously crash due to the strict mode

Reply:
To validate, try fetching form list with the current PR 2-3 times. You'll notice that second time, the crash will occur.

Then replace `HttpClientConnection` with `OkHttpConnection` in `AppDependencyModule` and perform the same tasks again. Now you won't notice any such crash

Reply:
@lognaturel when will we fix #2869?

Reply:
We can remove the `.penaltyDeath()` to prevent app crash and only display the error on the log if that PR is going to be merged in future

Reply:
Afraid there's no timeline for #2869 at the moment.

Reply:
What's blocking it?

Reply:
To answer your old question, it was blocked on getting some tests in place which @seadowg did!